### PR TITLE
Auto connect ports

### PR DESCRIPTION
**Pull request by odeimaiz**
## What do these changes do?
- [x] Refactor and apply good practices in NodeUI
- [x] When creating node connections, iterate over output ports and connect them to first compatible input port

## Related issue number
related to #1204 

AutoConnect

AutoConnect

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/web/client/source/class/osparc/component/workbench/NodeUI.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ qx.Class.define("osparc.component.workbench.NodeUI", {
   construct: function(node) {
     this.base();
 
+    this.setLayout(new qx.ui.layout.VBox());
```
Comment (oetiker) [inline services/web/client/source/class/osparc/component/workbench/NodeUI.js]: why not just add tis as `layout: new qx.ui.layout.VBox(),` to the next line ?

File: services/web/client/source/class/osparc/component/workbench/NodeUI.js [hunk 2]
```diff
@@ -107,44 +104,56 @@ qx.Class.define("osparc.component.workbench.NodeUI", {
       return this.getNode().getMetaData();
     },
 
-    __createNodeLayout: function() {
-      this.setLayout(new qx.ui.layout.VBox());
+    _createChildControlImpl: function(id) {
```
Comment (oetiker) [inline services/web/client/source/class/osparc/component/workbench/NodeUI.js]: the implementation depends on the call order ... this is not ideal ... by using a `grid` layout this problem could be avoided by inserting the individual elements into respective 'rows'.

File: services/web/client/source/class/osparc/component/workbench/NodeUI.js [hunk 3]
```diff
@@ -107,44 +104,56 @@ qx.Class.define("osparc.component.workbench.NodeUI", {
...
+          }
+          this.add(control);
+          break;
+        }
+        case "progress":
+          control = new qx.ui.indicator.ProgressBar().set({
+            height: 10,
+            margin: 4
+          });
+          this.add(control);
```
Comment (oetiker) [inline services/web/client/source/class/osparc/component/workbench/NodeUI.js]: calling `this.add(control)` at the end of the switch would do fine.

File: services/web/client/source/class/osparc/data/model/Node.js [hunk 4]
```diff
@@ -532,20 +532,24 @@ qx.Class.define("osparc.data.model.Node", {
...
         }
         this.__retrieveInputs();
       }
     },
 
-    createAutomaticPortConns: function(node1, node2) {
+    // Iterate over output ports and connect them to first compatible input port
+    __createAutoPortConnection: function(node1, node2) {
       // create automatic port connections
       console.log("createAutomaticPortConns", node1, node2);
```
Comment (ignapas) [inline services/web/client/source/class/osparc/data/model/Node.js]: Remove debugging ``console.log``

File: services/web/client/source/class/osparc/component/form/Auto.js [hunk 5]
```diff
@@ -603,18 +603,11 @@ qx.Class.define("osparc.component.form.Auto", {
...
-        fromPortLabel = fromPortId;
-      }
-      if (fromNodeLabel && fromPortLabel) {
-        this.getControlLink(toPortId).setValue("Linked to " + fromNodeLabel + ": " + fromPortLabel);
-      } else {
-        this.getControlLink(toPortId).setValue("Linked to " + fromNodeId + ": " + fromPortId);
-      }
+      const fromPortLabel = port ? port.label : null;
+      fromNode.bind("label", this.getControlLink(toPortId), "value", {
+        converter: label => label + ": " + fromPortLabel
```
Comment (ignapas) [inline services/web/client/source/class/osparc/component/form/Auto.js]: Would be cool to display the name of the selected file? Is it feasible?

File: services/web/client/source/class/osparc/store/Data.js [hunk 6]
```diff
@@ -50,7 +50,7 @@ qx.Class.define("osparc.store.Data", {
 
     getLocationsCached: function() {
       const cache = this.__locationsCached;
```
Comment (oetiker) [inline services/web/client/source/class/osparc/store/Data.js]: how about just initializing the _locationsCached at object creation time ?

**Pull request merged**